src/lib/Util.ts

```diff
@@ -269,20 +269,31 @@ export class Util {
     }
 
```

Comment:
MAM! Fajna zagadka! Nie mogłem odpuścić. Napisałem to w samym JSie i dopiero potem przekształciłem na TS więc nie wiem czy TS mi wyszedł, ale wymodziłem coś takiego:

```suggestion
public static convertFlatObjectToNestedObject(flatObject: any){
  return Object.entries(flatObject)
    .reduce((nestedObject, [field, value]) => applyFieldAsObject(nestedObject, field.split('.'), value), {})
}

public static applyFieldAsObject (target: any, fieldsStructure: string[], value: string | boolean | number) {
  const [firstField, ...rest] = fieldsStructure;
  const currentProperty = target ? target : {};

  if(rest.length === 0) {
    return { ...currentProperty, [firstField]: value };
  }

  return { 
    ...currentProperty, 
    [firstField]: reduceFields(currentProperty[firstField], rest, value)
  };
}
```

Nie wiem czy to jest lepsze, ale przyznam szczerze że to co jest obecnie czytało mi się źle i nie byłem pewien co to robi - ani nazwa ani kod mi tego w łatwy sposób nie mówił. Może w tym co zrobiłem też bym pomodził nad zmianą nazewnictwa. Nie wiem. Ale wydaje mi się że tę wersję łatwiej zrozumieć. Mniej się na raz dzieje, unikamy pętli w pętli i patrzymy na mniejszy kontekst. Co o tym myślicie @bartoszsuchocki @adrianlukaszek1992

Reply:
Tiaaa, applyFieldAsObject jest zdecydowanie złą nazwą. 
Target może można by nazwać targetObject? Nie wiem. Po rozwiązaniu zagadki nie mam weny na nazewnictwo.
currentProperty też mi nie pasuje ale to może dlatego że cała ta trójka po prostu ze sobą nie gra (nazwa funkcji, parametr i zmienna każde jest z innego koszyka)

Reply:
Bardzo fajne rozwiązanie :D